client/src/globals.ts

```diff
@@ -177,3 +188,19 @@ export function updateApiPrefix(): void {
   // path segments taken from the pathname.
   API.prefix = API.prefix.replace(REGEX_PATHNAME, pathname);
 }
+
+/**
+ * Update the base API URL for the current dataset using the S3 URI of the current dataset.
+ * @param s3Uri S3 URI of the current dataset.
+ * @returns The old API prefix.
+ */
+export function updateAPIWithS3(s3URI: S3URI): string {
+  if (!API) {
+    throw new Error("Unable to set API route.");
+  }
+  const oldAPI = API.prefix;
+  const URISafeS3URI = encodeURIComponent(s3URI);
+  const flaskSafeS3URI = `s3_uri/${encodeURIComponent(URISafeS3URI)}/`;
```

Comment:
Needs to be double encoded

Reply:
I think it's worth adding this as a comment and explains why it has to be double encoded 🙆‍♂️

Reply:
```suggestion
  // must be double quoted so slashes are not decoded early by flask WSGI.
  const URISafeS3URI = encodeURIComponent(s3URI);
  const flaskSafeS3URI = `s3_uri/${encodeURIComponent(URISafeS3URI)}/`;
```